packages/core/src/routes/admin-user.ts

```diff
@@ -194,31 +199,59 @@ export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
         body,
       } = ctx.guard;
 
-      await findUserById(userId);
+      const user = await findUserById(userId);
       await checkIdentifierCollision(body, userId);
 
+      const { roleNames, ...userUpdates } = body;
+
       // Temp solution to validate the existence of input roleNames
-      if (body.roleNames?.length) {
-        const { roleNames } = body;
+      if (roleNames) {
         const roles = await findRolesByRoleNames(roleNames);
 
-        if (roles.length !== roleNames.length) {
-          const resourcesNotFound = roleNames.filter(
-            (roleName) => !roles.some(({ name }) => roleName === name)
+        // Insert new roles
+        const newRoles = roleNames.filter((roleName) => !user.roleNames.includes(roleName));
```

Comment:
should we force existing roles here? to prevent typo

Reply:
or to use some explicit flag / params to create new roles when not exists

Reply:
We DO force existing roles (LINE 220), here the "insert" is about the relation table, `user-role`